contracts/src/spaces/facets/entitlements/extensions/EntitlementDataQueryable.sol

```diff
@@ -67,38 +70,56 @@ contract EntitlementDataQueryable is
     bytes32 channelId,
     string calldata permission
   ) internal view returns (Role[] memory) {
+    // retrive the roles associated with the channel
     uint256[] memory channelRoles = ChannelService.getRolesByChannel(channelId);
     uint256 channelRolesLength = channelRoles.length;
 
-    uint256 roleCount = 0;
+    // initialize arrays to store the matching role IDs
     uint256[] memory matchedRoleIds = new uint256[](channelRolesLength);
+    uint256 matchedRoleCount = 0;
 
-    RolesStorage.Layout storage ds = RolesStorage.layout();
+    // access roles storage layout
+    RolesStorage.Layout storage rs = RolesStorage.layout();
 
-    // Count the number of roles that have the requested permission and record their ids.
+    // iterate through channel roles and check for the requested permission
     for (uint256 i; i < channelRolesLength; i++) {
       uint256 roleId = channelRoles[i];
 
-      RolesStorage.Role storage role = ds.roleById[roleId];
+      RolesStorage.Role storage role = rs.roleById[channelRoles[i]];
 
+      // skip immutable roles
       if (role.isImmutable) {
         continue;
       }
 
-      // Check if the role has the requested permission.
-      if (role.permissions.contains(permission)) {
-        matchedRoleIds[roleCount] = roleId;
-        roleCount++;
+      bool hasPermission = false;
+
+      // check if role is associated with the channel and has the requested permission
+      if (rs.channelsByRole[roleId].contains(channelId)) {
+        StringSet.Set storage permissions = rs.permissionByChannelIdByRoleId[
```

Comment:
nonblocking comment: If possible I'd love for `permissionByChannelIdByRoleId` to be renamed to something that specifies these are override permissions - with the word `override` in the naming, so the code is a little more self-documenting, and it's more obvious why we would privilege this permission set and return on line 106.

Not sure what to suggest through, I think `overridePermissionsByChannelIdByRoleId` is probably a bit too long.

Reply:
done.